olive/workflows/run/run.py

```diff
@@ -44,7 +62,37 @@ def automatically_insert_passes(config):
     return new_engine, new_config
 
 
-def run(config: Union[str, Path, dict]):
+def dependency_setup(config):
+    required_packages = []
+
+    if config.engine.host.type == SystemType.Local:
+        required_packages.extend(
+            DEPENDENCY_MAPPING["device"][config.engine.host.type][config.engine.host.config.device]
+        )
+    else:
+        required_packages.extend(DEPENDENCY_MAPPING["device"][config.engine.host.type])
+
+    for _, pass_config in config.passes.items():
```

Comment:
The pass dependencies are only required in the local environment if the host for pass is local system. 
You can get the pass' host by something like `pass_config.get("host") or config.engine.host or LocalSystem()` (If the pass doesn't have host option set, it uses engine host. Default engine host is local system.

I think you can install the pass dependencies if type is local, otherwise, just print the dependencies and tell the user to make sure these packages are present in their aml/docker environment.

Reply:
change to only install pass related dependencies if it's local target

Reply:
In the latest change, I see that you check for the engine host. However, the host for a pass might be different from the engine host if the user provides a separate host for the pass. 

Could you check for the host per pass using the command I shared above? If the local, you can add to the packages list else print for the pass.

Reply:
I made it into two package lists: packages need to be installed locally and packages needed for remote host. For each pass, it will check pass.host, engine.host and local system in order.